pkg/kubectl/resource_printer.go

```diff
@@ -599,6 +615,11 @@ func printPod(pod *api.Pod, w io.Writer, options PrintOptions) error {
 func printPodBase(pod *api.Pod, w io.Writer, options PrintOptions) error {
 	name := pod.Name
 	namespace := pod.Namespace
+	kind := options.KindName
```

Comment:
If there is already a print func per type, would it be reasonable to infer the resourceName and call `ResourceShortFormFor` here? I'm guessing the kind is passed in so you can set kindname for thirdparty resources, but the custom resource printer is just dumping obj.Data anyway, so maybe it doesn't need to use the same pattern.